lib/engine.js

```diff
@@ -397,10 +397,11 @@ module.exports = function (s, conf) {
             trade_balance = n(s.balance.currency).divide(100).multiply(buy_pct)
             tradeable_balance = n(s.balance.currency).divide(100 + fee).multiply(buy_pct)
             expected_fee = n(trade_balance).subtract(tradeable_balance).format('0.00000000', Math.ceil) // round up as the exchange will too
+            console.log(`s.product.asset_increment: ${s.product.asset_increment}`)
             if (buy_pct + fee < 100) {
-              size = n(tradeable_balance).divide(price).format('0.00000000')
+              size = n(tradeable_balance).divide(price).format(s.product.asset_increment ? s.product.asset_increment : '0.00000000', Math.floor)
```

Comment:
I think we should define what asset_increment is.  I can see its usefulness in other exchanges, and we should update those for consistency.
But we do need the definition at least as this property is now taken into account by the engine.

What is the purpose, and some sample values if the exchange doesn't directly return it should it be 0 or the same as increment

Reply:
I have no knowledge or experience with the way other exchanges handle this, and I seem to be the only one who has been bitten by this bug before so apparently the previous behavior hasn't bothered anyone. That's why this conditional preserves the previous behavior for all other scenarios.

Reply:
Fee really is the only thing which should get rounded differently, leave the math to the bot and numbro as this messes with our previous calculation.

If the fix is "exchange specific" consider implementing it like e.g. `s.exchange.makerBuy100Workaround` instead of changing the default behavior.

Reply:
It shouldn't be exchange specific. It should be "coin that isn't 8 precision" specific, which lots of exchanges have. I just don't know how that's handled on all other exchanges so I've done this to create a starting point for it.

Reply:
So its exchange-coin-specific?

In that case do the Math.floor only if s.product.asset_increment exists, as we expect all coins to have a precision of 8 by default.